src/selectors/visibleSelectedFrame.js

```diff
@@ -2,26 +2,30 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
+// @flow
+
 import { getSelectedLocation } from "../reducers/sources";
 import { getSelectedFrame } from "../reducers/pause";
 import { isOriginalId } from "devtools-source-map";
+import { createSelector } from "reselect";
 
 function getLocation(frame, isGeneratedSource) {
   return isGeneratedSource
     ? frame.generatedLocation || frame.location
     : frame.location;
 }
 
-export default function getVisibleSelectedFrame(state: OuterState) {
-  const selectedLocation = getSelectedLocation(state);
-  const isGeneratedSource = !isOriginalId(selectedLocation.sourceId);
-  const selectedFrame = getSelectedFrame(state);
-
-  if (!selectedFrame) {
-    return;
+const getVisibleSelectedFrame = createSelector(
+  getSelectedLocation,
+  getSelectedFrame,
+  (selectedLocation, selectedFrame) => {
+    const isGeneratedSource = !isOriginalId(selectedLocation.sourceId);
+    return !selectedFrame
+      ? null
+      : {
+          location: getLocation(selectedFrame, isGeneratedSource)
+        };
   }
+);
 
-  return {
-    location: getLocation(selectedFrame, isGeneratedSource)
-  };
-}
+export default getVisibleSelectedFrame;
```

Comment:
so... everytime we have an issue where we need to use `reselect` to prevent selectors from returning new objects every time I think of @asolove and am grateful for his help this past spring figuring this out.

This issue was introduced when @assafdd moved the debug line to using this new selector so that we grab the correct location based on the selected location. 

It's really hard to detect when a selector change causes a component to re-render aggressively (like 20 times). We should think about a monitoring strategy so we're notified.